ocaml/xapi/repository_helpers.ml

```diff
@@ -890,3 +962,32 @@ let consolidate_updates_of_host ~updates_info host updates_of_host =
       ("updates", `List (List.map (fun uid -> `String uid) (UpdateIdSet.elements uids)))]
   in
   (json_of_host, uids)
+
+let append_by_key l k v =
```

Comment:
Is it legal to have a `(key, [v;v;])`: the same value is added multiple times to `key`?

Reply:
The final usage of this list `[v;v]` is to check if it is empty or not. So same value being added multiple times will not get wrong result.